lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -32,20 +32,97 @@
-import numpy as np
-import cf_units
+from iris.tests import IrisTest
+from iris.util import squeeze
 
 from improver.nowcasting.lightning import NowcastLightning as Plugin
 from improver.utilities.cube_checker import find_dimension_coordinate_mismatch
-from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
-    set_up_cube, set_up_cube_with_no_realizations)
-from improver.tests.ensemble_calibration.ensemble_calibration.helper_functions\
-    import add_forecast_reference_time_and_forecast_period
+from improver.tests.set_up_test_cubes import (
+    set_up_variable_cube, set_up_probability_cube)
+
+
+def set_up_lightning_test_cubes(validity_time=dt(2015, 11, 23, 7),
+                                fg_frt=dt(2015, 11, 23, 3),
+                                grid_points=3):
+    """Set up five cubes for testing nowcast lightning.
+
+    The cube coordinates look like this:
+        Dimension coordinates:
+            projection_y_coordinate: grid_points;
+            projection_x_coordinate: grid_points;
+        Scalar coordinates:
+            time: 2015-11-23 07:00:00
+            forecast_reference_time: 2015-11-23 07:00:00
+            forecast_period: 0 seconds
+
+    Kwargs:
+        grid_points (int):
+            Number of points along each spatial axis (square grid)
+        validity_time (datetime.datetime):
+            Time to use for test cubes
+        fg_frt (datetime.datetime):
+            Forecast reference time for first_guess_cube, which needs
+            to have different forecast periods for different tests
+
+    Returns:
+        template_cube (iris.cube.Cube)
+        first_guess_cube (iris.cube.Cube)
+        lightning_rate_cube (iris.cube.Cube)
+        prob_precip_cube (iris.cube.Cube):
+            Has extra coordinate of length(3) "threshold" containing
+            points [0.5, 7., 35.] mm h-1
+        prob_vii_cube (iris.cube.Cube):
+            Has extra coordinate of length(3) "threshold" containing
+            points [0.5, 1., 2.] kg m-2
+    """
+    # template cube full of ones with one zero point
+    data = np.ones((grid_points, grid_points), dtype=np.float32)
+    template_cube = set_up_variable_cube(
+        data.copy(), name='lightning_rate', units='min-1',
+        time=validity_time, frt=validity_time, spatial_grid='equalarea')
+    template_cube.data[1, 1] = 0.
+
+    # first guess lightning rate probability cube with flexible forecast
+    # period (required for level 2 lighting risk index)
+    prob_fg = np.array([data.copy()], dtype=np.float32)
+    first_guess_cube = set_up_probability_cube(
+        prob_fg, np.array([0], dtype=np.float32), threshold_units='s-1',
+        variable_name='lightning_rate', time=validity_time, frt=fg_frt,
+        spatial_grid='equalarea')
+    first_guess_cube = squeeze(first_guess_cube)
+
+    # lightning rate cube full of ones
+    lightning_rate_cube = set_up_variable_cube(
+        data.copy(), name='rate_of_lightning', units='min-1',
```

Comment:
The template cube is weird.  It just provides the coordinate metadata for the output cube.  I might redesign it to be a probability cube (ie a copy of `fg_cube`) in this case.

However in the plugin:

- The "template cube" equivalent is actually provided by the precip probability cube - so it's pretty arbitrary what this cube needs to look like
- The cube names expected by `process` are "**rate_of_lightning**" (for `self.ltng_cube`) and "probability_of_**lightning_rate**_above_threshold" (for `self.fg_cube`).  There's potential for these to be wrong.  @MoseleyS can you comment?

Reply:
Umm..  Firstly, the unit-tests I wrote only cared about meta-data that was actually used in the function being tested (in hindsight, this is perhaps unhelpful). For cube names, I think we should be more consistent here and make our variable name "rate_of_lightning", so that the derivative becomes "probability_of_rate_of_lightning_above_threshold".

Reply:
OK.  I will update both the plugin and the unit tests in light of this.  I am also going to rename the template cube to something totally unrelated to lightning, for clarity that this name doesn't matter.